provider/osio/osio.go

```diff
@@ -0,0 +1,178 @@
+package osio
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/cenk/backoff"
+	"github.com/containous/traefik/job"
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/provider"
+	"github.com/containous/traefik/safe"
+	"github.com/containous/traefik/types"
+)
+
+const (
+	providerName  = "osio"
+	authorization = "Authorization"
+)
+
+// Provider holds configurations of the provider.
+type Provider struct {
+	provider.BaseProvider `mapstructure:",squash" export:"true"`
+
+	RefreshSeconds       int    `description:"Polling interval (in seconds)" export:"true"`
+	ServiceAccountID     string `description:"Service Account ID" export:"true"`
+	ServiceAccountSecret string `description:"Service Account Secret" export:"true"`
+	TokenAPI             string `description:"Auth token API" export:"true"`
+	ClusterAPI           string `description:"Cluster data API" export:"true"`
+
+	client            client
+	tokenResp         *tokenResponse
+	defaultBackendURL string
+}
+
+// Provide allows the osio provider to provide configurations to traefik
+// using the given configuration channel.
+func (p *Provider) Provide(configChan chan<- types.ConfigMessage, pool *safe.Pool, constraints types.Constraints) error {
+	log.Debugf("Configuring %s provider", providerName)
+	p.init(configChan)
+	p.scheduleConfigPull(configChan, pool)
+	return nil
+}
+
+func (p *Provider) scheduleConfigPull(configChan chan<- types.ConfigMessage, pool *safe.Pool) {
```

Comment:
is this `pull` or `poll` or `pool` I am confused here, also a comment would be helpful?

Reply:
now it's changed to schedule()